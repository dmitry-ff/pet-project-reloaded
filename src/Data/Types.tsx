export type TUserData = {
  data: {
    labels: string[];
    datasets: {
      label: string;
      data: number[];
      backgroundColor: CanvasPattern[];
      borderColor: string[];
      borderWidth: number;
      hoverOffset: number;
    }[];
  }
};
export type TLineData = {
  data: {
    labels: string[];
    datasets: {
      label: string;
      data: number[];
      borderColor: string;
      backgroundColor: string;
    }[];
  }
};

export type TLoading = {
  loading: boolean;

};
export type TContext = [
  context: boolean,
  setContext: React.Dispatch<React.SetStateAction<boolean>>,
];
export type TData = {
  date: string,
  day: {
    maxtemp_c: number,
    mintemp_c: number,
    avgtemp_c: number,
    maxwind_kph: number,
    daily_chance_of_rain: number,
    daily_chance_of_snow: number,
    condition: {
      text: string,
      icon: string
    }
  }
}

export type TResponse = {
  forecast: {
    forecastday:
    {
      date: string,
      day: {
        maxtemp_c: number,
        mintemp_c: number,
        avgtemp_c: number,
        maxwind_kph: number,
        daily_chance_of_rain: number,
        daily_chance_of_snow: number,
        condition: {
          text: string,
          icon: string
        }
      },
    }[]
  }
}