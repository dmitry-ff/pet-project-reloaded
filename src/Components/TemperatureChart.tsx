// import _ from "lodash";
// import React from "react";
// import { MONTH_S } from "../Data/Consts";
// import { Chart as ChartJS } from 'chart.js/auto';
// import { Line } from 'react-chartjs-2';
// import { TLoading, TUserData } from '../Data/Types';

// function TemperatureChart({ loading, data }: TLoading & TUserData) {

//   const [chartData, setData] = React.useState({
//     labels: _.map(data, item => {
//       return `${MONTH_S[new Date(item.date).getMonth()]}, ${new Date(item.date).getDate()}`
//     }),
//     datasets: [
//       {
//         label: 'max t, °C',
//         data: _.map(data, item => {
//           return item.day.maxtemp_c;
//         }),
//         backgroundColor: 'rgba(222, 99, 132, 0.2)',
//         borderColor: 'rgba(255, 99, 132, 1)',
//         borderWidth: 2,
//       },
//       {
//         label: 'avg t, °C',
//         data: _.map(data, item => {
//           return (item.day.maxtemp_c + item.day.mintemp_c) / 2;
//         }),
//         backgroundColor: 'rgba(48, 207, 113,0.2)',
//         borderColor: 'rgba(48, 207, 113,1)',
//         borderWidth: 2,
//       },
//       {
//         label: 'min t, °C',
//         data: _.map(data, item => {
//           return item.day.mintemp_c;
//         }),
//         backgroundColor: 'rgba(1, 99, 132, 0.2)',
//         borderColor: 'rgba(1, 99, 132, 1)',
//         borderWidth: 2,
//       },
//     ],
//   });

//   return <>{!loading && <Line data={chartData} />
//   }</>
// }

// export default TemperatureChart;
export { };