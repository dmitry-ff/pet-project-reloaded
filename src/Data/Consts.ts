export const WEEK_S: string[] = ['Sun', 'Mon', 'Tue', 'Wed', 'Thu', 'Fri', 'Sat'];
export const MONTH_S: string[] = ['Jan', 'Feb', 'Mar', 'Apr', 'May', 'Jun', 'Jul', 'Aug', 'Sep', 'Oct', 'Nov', 'Dec'];
export const CHART_NAMES: string[] = ['BarChart', 'LineChart', 'DoughnutChart', 'PieChart'];