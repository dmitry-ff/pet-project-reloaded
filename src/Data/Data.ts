import pattern from 'patternomaly';
import { draw } from 'patternomaly';


export const UserData = {

  labels: ['Red', 'Blue', 'Yellow',],
  datasets: [
    {
      label: '# of Votes',
      data: [12, 19, 3, 5, 2, 3],
      backgroundColor: [
        pattern.draw('triangle', '#adbe41'),
        pattern.draw('ring', '#9e73bb'),
        pattern.draw('weave', '#17becf'),
        pattern.draw('diamond', '#2ca02c'),
        pattern.draw('zigzag-vertical', '#de215e'),
        pattern.draw('box', 'rgb(255, 99, 132, 0.4)')
      ],
      borderColor: [
        'rgba(255, 99, 132, 1)',
        'rgba(54, 162, 235, 1)',
        'rgba(255, 206, 86, 1)',
        'rgba(75, 192, 192, 1)',
        'rgba(153, 102, 255, 1)',
        'rgba(255, 159, 64, 1)',
      ],
      borderWidth: 1,
      hoverOffset: 3,
    },
  ],
}