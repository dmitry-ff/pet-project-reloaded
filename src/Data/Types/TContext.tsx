export type TContext = {
  context: boolean;
  setContext: React.Dispatch<React.SetStateAction<boolean>>;
};
