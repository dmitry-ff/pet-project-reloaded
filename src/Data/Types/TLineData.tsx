export type TLineData = {
  data: {
    labels: string[];
    datasets: {
      label: string;
      data: number[];
      borderColor: string;
      backgroundColor: string;
    }[];
  };
};
