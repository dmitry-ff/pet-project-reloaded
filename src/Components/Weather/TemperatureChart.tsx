import _ from "lodash";
import React from "react";
import { MONTH_S } from "../../Data/Consts";
import { Line } from 'react-chartjs-2';
import { TLoading } from '../../Data/Types';
import { EmotionJSX } from "@emotion/react/types/jsx-namespace";
type TData = {
  data: {
    date: string,
    day: {
      maxtemp_c: number,
      mintemp_c: number,
      avgtemp_c: number,
      maxwind_kph: number,
      daily_chance_of_rain: number,
      daily_chance_of_snow: number,
      condition: {
        text: string,
        icon: string
      }
    }
  }[]
}
const TemperatureChart: React.FC<TLoading & TData> = ({ loading, data }): EmotionJSX.Element => {

  const [chartData, setData] = React.useState({
    labels: _.map(data, item => {
      return `${MONTH_S[new Date(item.date).getMonth()]}, ${new Date(item.date).getDate()}`
    }),
    datasets: [
      {
        label: 'max t, °C',
        data: _.map(data, item => {
          return item.day.maxtemp_c;
        }),
        backgroundColor: 'rgba(222, 99, 132, 0.2)',
        borderColor: 'rgba(255, 99, 132, 1)',
        borderWidth: 2,
        tension: 0.3
      },
      {
        label: 'avg t, °C',
        data: _.map(data, item => {
          return (item.day.maxtemp_c + item.day.mintemp_c) / 2;
        }),
        backgroundColor: 'rgba(48, 207, 113,0.2)',
        borderColor: 'rgba(48, 207, 113,1)',
        borderWidth: 2,
        tension: 0.3

      },
      {
        label: 'min t, °C',
        data: _.map(data, item => {
          return item.day.mintemp_c;
        }),
        backgroundColor: 'rgba(1, 99, 132, 0.2)',
        borderColor: 'rgba(1, 99, 132, 1)',
        borderWidth: 2,
        tension: 0.3
      },
    ],
  });

  return <>{!loading && <Line data={chartData} />
  }</>
}

export default TemperatureChart;
// export { };