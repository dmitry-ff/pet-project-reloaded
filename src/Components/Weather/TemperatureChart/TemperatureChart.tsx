import React, { useEffect } from "react";
import { EmotionJSX } from "@emotion/react/types/jsx-namespace";
import { MONTH_S } from "../data/MONTH_S";
import { Line } from 'react-chartjs-2';
import { TData } from "../types/TData";
import { newLineData } from "../../Charts/LineChart/types/TLineData";
import { dataTransform } from "./utils/dataTransform";

const TemperatureChart = (props: { loading: boolean, data: TData[] }): EmotionJSX.Element | null => {
  const [chartData, setData] = React.useState<newLineData | null>(null);
  const { loading, data } = props;
  useEffect(() => {
    setData(dataTransform(data));
  }, [data])

  if (!loading && chartData) {
    return <Line data={chartData} />
  }
  return null;
}

export default TemperatureChart;
// export { };