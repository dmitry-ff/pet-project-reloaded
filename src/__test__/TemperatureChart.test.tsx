import { render, shallow } from "enzyme"
import { TData } from "../Data/Types"
import TemperatureChart from '../Components/Weather/TemperatureChart'

type TProps = {
  loading: boolean,
  data: TData[]
};


const props = {
  loading: true,
  data: [
    {
      date: '2022-02-21',
      day: {
        maxtemp_c: 11,
        mintemp_c: 11,
        avgtemp_c: 11,
        maxwind_kph: 11,
        daily_chance_of_rain: 11,
        daily_chance_of_snow: 11,
        condition: {
          text: 'asdasdasd',
          icon: 'asdasdasd',
        }
      }
    }
  ]
};
//const setUp = (props: TProps) => render(<TemperatureChart {...props} />)

it('should render component', () => {

  // const view = setUp(props);
  const view = render(<TemperatureChart {...props} />)

})

export { }