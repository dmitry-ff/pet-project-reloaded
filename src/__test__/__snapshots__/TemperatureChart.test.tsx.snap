// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Temperature chart component should render canvas element with data array 1`] = `
<canvas
  height={150}
  role="img"
  width={300}
/>
`;

exports[`Temperature chart component should render canvas element with empty data array 1`] = `
<canvas
  height={150}
  role="img"
  width={300}
/>
`;
